src/blueapi/worker/worker.py

```diff
@@ -1,27 +1,73 @@
 from abc import ABC, abstractmethod
-from typing import Generic, TypeVar
+from typing import Generic, List, TypeVar
 
 from blueapi.core import DataEvent, EventStream
+from blueapi.utils import BlueapiBaseModel
 
 from .event import ProgressEvent, WorkerEvent
 
 T = TypeVar("T")
 
 
+class TrackableTask(BlueapiBaseModel, Generic[T]):
+    """
+    A representation of a task that the worker recognizes
+    """
+
+    task_id: str
+    task: T
+    is_complete: bool = False
+    is_error: bool = False
```

Comment:
I moved it here because it has become part of the `Worker` API, rather than just for internal use by `RunEngineWorker`. So one of the API methods on `Worker` returns a `TrackableTask`, which means things outside need to know about `TrackableTask`. Ideally, because it's an interface, everything in `worker.py` should not know about anything else, other things should just know about it.